lib/compeon/rack_tools/pipes/parse_token_with_key.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require 'compeon/rack_tools/http_errors'
+require 'compeon/token'
+
+module Compeon
+  module RackTools
+    module Pipes
+      PARSE_TOKEN_WITH_KEY = lambda do |key:|
+        lambda do |request:, **rest|
+          token_string = request.env['HTTP_AUTHORIZATION']&.match(/^token (?<token>.*)/)&.named_captures&.[]('token')
```

Comment:
I think this is much more readable, would you agree? (It is not so clever, though)
```suggestion
          auth_header = request.env['HTTP_AUTHORIZATION'] || ''
          token_string = auth_header[/^token (.*)/, 1]
```

Reply:
Actually it's a copy pasta from `PARSE_TOKEN` and I just replaced what was necessary :D

Reply:
Or, if you insist on keeping it a one-liner
```ruby
token_string = request.env.fetch('HTTP_AUTHORIZATION', '')[/^token (.*)/, 1]
```

Reply:
```
token_string = request.env.fetch('HTTP_AUTHORIZATION', '')[/^token (?<token>.*)/, 'token']
```

Reply:
@timoschilling I considered that, but it seems kind of obvious that the capture group is a token, given the name of the variable and the leading `^token`. I am  not sure whether we need the word three times in a row 😉.

Reply:
I'm also ok with @lagr solution ;)